packages/compass-preferences-model/src/in-memory-storage.ts

```diff
@@ -0,0 +1,35 @@
+import { getDefaultsForStoredPreferences } from './preferences-schema';
+import type { AllPreferences, StoredPreferences } from './preferences-schema';
+
+export interface BasePreferencesStorage {
+  setup(): Promise<void>;
+  getPreferences(): StoredPreferences;
+  updatePreferences(attributes: Partial<StoredPreferences>): Promise<void>;
+}
+
+export class InMemoryStorage implements BasePreferencesStorage {
```

Comment:
this is a mega NIT, probably unrelated and on code that was already there, however i find `BasePreferencesStorage` not really idiomatic here.

`BasePreferencesStorage` makes me think of a base abstract class that is incomplete and that we are extending. Here instead we are providing a completely new implementation of an interface, if that is the case i would rather expect the interface name to be something like `PreferencesStorage` instead of `BasePreferencesStorage`.